### PR TITLE
lobby and match making server now our own + baby carrot state machine

### PR DESCRIPTION
**Pull request by joshuakguo**
- Moving over to our own lobby and match making server
- Baby carrot AI is now a state machine
- Previous behavior is a ROAM state and there is a new HOLD state
- non hosts now no longer initialize their AI Controller

it created this Android.mk file, idk what it does, hopefully nothing

### CODE REVIEW

File: source/controllers/AIController.cpp [hunk 1]
```diff
@@ -17,21 +17,61 @@ bool AIController::init(std::shared_ptr<Map> &map) {
     return true;
 }
 
+void AIController::updateBabyCarrotState(const std::shared_ptr<BabyCarrot> &babyCarrot) {
+    if (!babyCarrot->hasTarget()) {
+        if (babyCarrot->getState() == State::HOLD) {
+            if (((float) std::rand()/ RAND_MAX) < 0.5) {
+                babyCarrot->setState(State::HOLD);
```
Comment (jffzhou) [inline source/controllers/AIController.cpp:24]: this line seems redundant

**Review by jffzhou — CHANGES_REQUESTED** [review]
it seems like since ```updateBabyCarrotState``` is called every time ```updateBabyCarrot``` is called that the states would be changing really rapidly, could you make it only called once every second (or something like that) instead?

also make sure to remove the build folder before merging

**Comment by joshuakguo** [comment]
removed build folder and changed state update to only when there's no target

**Review by joshuakguo — COMMENTED** [review]
changes made

**Review by joshuakguo — COMMENTED** [review]
im stupid

**Pull request merged**